lib/marc/xml_parsers.rb

```diff
@@ -176,7 +194,15 @@ def each
           event = @parser.pull
           # if it's the start of a record element 
           if event.start_element? and strip_ns(event[0]) == 'record'
-            yield build_record
+            record = build_record
+
+            if @record[:error]
+              warn "Skipping record do to error: #{@record[:error]}"
+            else
```

Comment:
We really have no logger object, the only thing we can do is write to stderr (which is what warn does)? That's unfortunate. Hmm.

Reply:
Yeah, I see a couple of places where there is a TODO regarding logger, but it doesn't seem like it's ever been implemented.  It would be nice to be able to configure a logger and grab that instead. Or if a no logger has not been configured than print to stderr.